src/inpage/QryptoRPCProvider.ts

```diff
@@ -1,7 +1,93 @@
-import { rpcCall } from './rpcCall';
+import { isEmpty } from 'lodash';
 
-export class QryptoRpcProvider {
-  public rawCall(method: string, args: any[]) {
-    return rpcCall(method, args);
+import { IRPCCallRequest, IRPCCallRequestPayload, IExtensionAPIMessage, IExtensionMessageData, IRPCCallResponsePayload } from '../types';
+import { TARGET_NAME, API_TYPE } from '../constants';
+import Config from '../config';
+import { generateRequestId } from '../utils';
+
+const { DEFAULT_AMOUNT, DEFAULT_GAS_LIMIT, DEFAULT_GAS_PRICE } = Config.TRANSACTION;
+
+export class QryptoRPCProvider {
+  private requests: { [id: string]: IRPCCallRequest } = {};
+
+  public rawCall = (method: string, args: any[]) => {
+    return new Promise((resolve, reject) => {
+      const id = this.trackRequest(resolve, reject);
+      this.postMessageToContentscript({
+        type: API_TYPE.RPC_REQUEST,
+        payload: { method, args, id },
+      });
+    });
+  }
+
+  public sendToContract = (
```

Comment:
i think we should avoid convenient functions on `QryptoRPCProvider` specifically.

would suggest moving `sendToContract` and `callContract` into `QryptoRPCHelper` or whatever if the convenience is required.